src/project/environment.rs

```diff
@@ -336,6 +355,27 @@ impl<'p> Environment<'p> {
     pub fn has_pypi_dependencies(&self) -> bool {
         self.features(true).any(|f| f.has_pypi_dependencies())
     }
+
+    /// Return a combination of static environment variables generated from the project and the environment
+    /// and from running activation script
+    pub async fn get_env_variables(&self) -> miette::Result<HashMap<String, String>> {
+        let mut locked = self.vars.lock().await;
+
+        if !locked.is_empty() {
```

Comment:
I think its never good practice (and very pythonic) to depend on something being empty as meaning that it wasnt set. Its better to wrap these things in an Option to provide more semantics to the type.

Additionally for this usecase you could use an async_once_cell.

Reply:
I agree!